quilt_repository_format.md

```diff
@@ -0,0 +1,137 @@
+# Quilt CLI Local Storage
+
+## Motivation
+The primary motivation for unifying the local package store is that any Python process will be able to access any locally installed Quilt package. A common point of friction for Quilt users has been that packages built on the command line aren't visible inside Jupyter notebooks if the Jupyter server is running in a different directory. Unifiying the local storage by adopting the Docker storage model solves that problem and also enables useful future features like comparing (diff) package versions and tracking revision history locally without pushing to a registry.
+
+## Directory Structure
+This change proposes changing from a hierarchy of separate local package stores(quilt_packages) like npm, to a single store like Docker.
+
+```bash
+BASE_DIR/objs/
+```
+Stores binary data objects identified by hash. These objects include compressed raw files and Parquet files. Object hashes are verified and objects are stored only once (deduplication).
+
+```bash
+BASE_DIR/pkgs/
+```
+
+Stores package manifest files (JSON format) for all package instances. Package manifests are identified by the hash of the package.
+
+```bash
+BASE_DIR/contents.json
+```
+
+Catalog of locally resident packages. The contents manfiest maps identifiers including package name, tags and versions to package instances.
+
+## Contents Manifest
+The contents manifest file ```contents.json``` contains a mapping of package names, versions and tags to locally installed package instances.
+
+```json
+<pkgname>:
+    tags:
+        tag:        <hash>
+    versions:
+        version:    <hash>
+```
+
+## Commands
+
+```bash
+quilt ls
+
+# Package       Version     Tag     ID      Created     Size
+user/package    <version>   <tag>   <hash>  <date>      <bytes on disk>
+```
+
+Lists all locally installed package instances. The output of ls shows the package name and any tags or versions associated with the instance.
+
+```bash
+quilt tag add <hash> <tag>
+quilt tag rm <hash> <tag>
+```
+
+Adds or removes a tag from a package instance.
+
+```bash
+quilt version add <hash> <tag>
+quilt version rm <hash> <tag>
+```
+
+Adds or removes a version from a package instance. Note: once pushed to a Quilt registry, there is no way to remove a version without deleting the package.
+
+```bash
+quilt push <package>
+```
+
+Pushes a package instance to the Quilt registry. Quilt push will record any local tags or versions to the registry.
+
+```bash
+quilt install <package>[:tag:version]
```

Comment:
The syntax is not clear to me. How do you specify tag vs version?

Reply:
I wasn't meaning to change the syntax for specifying versions and tags on install.